From f5444cc996e931c10fe55969c775941b63c9a51c Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 5 Dec 2016 14:18:04 +0200
Subject: [PATCH 0635/1345] net: mvpp2x: change TXQ ring size, CPU chunk, aggr
 TXQ size parameters

commit  fefee357a4f60380b5b58887f14ef3cdade12e5a from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Increase TXQ ring to 2048 descriptors
- Increase CPU chunk to 128 descriptors
- Increase aggr TXQ ring to 512
- Set default TXQ done coalescing 1000 usec
- Increase number of external buffers to total size of skb
  shadow ring

Change-Id: I2783b86ef56b89359356fc6149f53df54c81cc07
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34349
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h      |    4 ++--
 .../net/ethernet/marvell/mvpp2x/mv_pp2x_hw_type.h  |    6 +++---
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
index 20d7914..ec6a312 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
@@ -137,7 +137,7 @@
 /* Coalescing */
 #define MVPP2_TXDONE_COAL_PKTS		64
 #define MVPP2_TXDONE_HRTIMER_PERIOD_NS	1000000UL
-#define MVPP2_TXDONE_COAL_USEC		10000
+#define MVPP2_TXDONE_COAL_USEC		1000
 
 #define MVPP2_RX_COAL_PKTS		32
 #define MVPP2_RX_COAL_USEC		64
@@ -181,7 +181,7 @@
 #define MVPP2_ETH_SHADOW_EXT		0x2
 
 #define MVPP2_EXTRA_BUF_SIZE	120
-#define MVPP2_EXTRA_BUF_NUM	MVPP2_MAX_TXD
+#define MVPP2_EXTRA_BUF_NUM	(MVPP2_MAX_TXD * MVPP2_MAX_TXQ)
 
 enum mvppv2_version {
 	PPV21 = 21,
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw_type.h b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw_type.h
index 1849b68..83d2d87 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw_type.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_hw_type.h
@@ -1221,13 +1221,13 @@
 #define MVPP2_MAX_RXD			1024
 
 /* Max number of Tx descriptors */
-#define MVPP2_MAX_TXD			1024
+#define MVPP2_MAX_TXD			2048
 
 /* Amount of Tx descriptors that can be reserved at once by CPU */
-#define MVPP2_CPU_DESC_CHUNK		64
+#define MVPP2_CPU_DESC_CHUNK		128
 
 /* Max number of Tx descriptors in each aggregated queue */
-#define MVPP2_AGGR_TXQ_SIZE		256
+#define MVPP2_AGGR_TXQ_SIZE		512
 
 /* Descriptor aligned size */
 #define MVPP2_DESC_ALIGNED_SIZE		32
-- 
1.7.9.5

